src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -683,9 +682,17 @@
              return;
         }
 
-        var objId = X[self.daoKey_].of.ID.type === 'Long' ? parseInt(this.objId) : this.objId;
-
-        return X[this.daoKey_]
+        var objId, daoKey;
+        if ( self.refObj ) {
+          var res = self.refObj.split(':');
+          daoKey = res[0];
+          objId = X[daoKey].of.ID.type === 'Long' ? parseInt(res[1]) : res[1];
+        } else {
+          objId = X[self.daoKey_].of.ID.type === 'Long' ? parseInt(self.objId) : self.objId;
```

Comment:
More generically, this could be done as X[self.daoKey_].of.ID.adapt(null, self.objId), then it would work for any type.

Reply:
ok will address, thanks